inc/Data/ThirdPartyDataFormatter.php

```diff
@@ -192,21 +216,29 @@ public static function formatUrl(string $url, array $params, array $args, array
     /**
      * Formats the given code arguments into a code string.
      *
-     * @see https://github.com/GoogleChromeLabs/third-party-capital/blob/0831b937a8468e0f74bd79edd5a59fa8b2e6e763/src/utils/index.ts#L48
+     * @see https://github.com/GoogleChromeLabs/third-party-capital/blob/54cd44d1bd197a7809ab2f6ede4d13a973087c3d/src/utils/index.ts#L52
      *
-     * @param string               $code Code string with placeholders for URL query parameters.
-     * @param array<string, mixed> $args Input arguments for the src attribute query parameters.
+     * @param string               $code           Code string with placeholders for URL query parameters.
+     * @param array<string, mixed> $args           Input arguments for the src attribute query parameters.
+     * @param array<string, mixed> $optionalParams Optional. Optional parameter names and their defaults.
+     *                                             Default empty array.
      * @return string HTML string.
      */
-    public static function formatCode(string $code, array $args): string
-    {
+    public static function formatCode(
+        string $code,
+        array $args,
+        array $optionalParams = []
+    ): string {
         return preg_replace_callback(
             '/{{([^}]+)}}/',
-            static function ($matches) use ($args) {
+            static function ($matches) use ($args, $optionalParams) {
                 if (isset($args[ $matches[1] ])) {
-                    return $args[ $matches[1] ];
+                    return json_encode($args[ $matches[1] ]);
+                }
+                if (isset($optionalParams[ $matches[1] ])) {
+                    return json_encode($optionalParams[ $matches[1] ]);
                 }
-                return '';
+                return '""';
```

Comment:
why does this return '""'?

Reply:
It's the same as `json_encode( '' )`, but since that's predictable, I just made it `'""'`. I can add a comment to clarify.

Reply:
Done in https://github.com/GoogleChromeLabs/third-party-capital/pull/70/commits/58de6d48a05f4e1a5994c910d96fd59894039338